docs/data-storage.md

```diff
@@ -72,6 +72,13 @@ fairly unstructured way. We mostly store transaction outcomes as a JSON blob,
 but some metadata is stored in separate columns to enable more efficient
 indexing.
 
+In the event of a protocol update, a new chain is started, with a new genesis
+block that is based on the last finalized block of the previous chain. The new
+chain maintains a separate **tree state** and **block state** from the previous
+chain, while the **transaction index** is shared. Once a protocol update
```

Comment:
It is worth mentioning that the block heights in the transaction index will start again at 0 upon a protocol update, at least as it stands now.

Reply:
Pending #148, the block heights will be absolute in the transaction index. I have added that detail in the description of the transaction index.